PagingWithNetworkSample/lib/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/paging/SynchronousRemoteStates.kt

```diff
@@ -0,0 +1,141 @@
+package com.android.example.paging.pagingwithnetwork.reddit.paging
+
+import androidx.paging.CombinedLoadStates
+import androidx.paging.LoadState
+import androidx.paging.LoadState.*
+import androidx.paging.LoadStates
+import com.android.example.paging.pagingwithnetwork.reddit.paging.SynchronousRemoteState.*
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.scan
+import kotlin.Error
+
+/**
+ * Track the combined [LoadState] of [RemoteMediator] and [PagingSource], so that each load type
+ * is only set to [NotLoading] when [RemoteMediator] load is applied on presenter-side.
+ */
+private class SynchronousRemoteStates {
+    var refresh: LoadState = NotLoading(endOfPaginationReached = false)
+        private set
+    var prepend: LoadState = NotLoading(endOfPaginationReached = false)
+        private set
+    var append: LoadState = NotLoading(endOfPaginationReached = false)
+        private set
+    private var refreshState: SynchronousRemoteState = NOT_LOADING
+    private var prependState: SynchronousRemoteState = NOT_LOADING
+    private var appendState: SynchronousRemoteState = NOT_LOADING
+
+    fun toLoadStates() = LoadStates(
+        refresh = refresh,
+        prepend = prepend,
+        append = append
+    )
+
+    internal fun updateFromCombinedLoadStates(combinedLoadStates: CombinedLoadStates) {
```

Comment:
add basic comments to this and below functions, so it's clear how it works

Reply:
Done